src/utils/writeToCsv.ts

```diff
@@ -0,0 +1,41 @@
+import { createFile, fileExists, readFile } from '../file'
+import { prefixMessage } from '../error'
+import stringify from 'csv-stringify/lib/sync'
+
+export const writeToCsv = async (
+  csvFileRealPath: string,
+  data: any,
+  columns: any
+) => {
+  if (
+    !(await fileExists(csvFileRealPath).catch((err) => {
+      throw prefixMessage(err, 'Error while checking if csv file exists.\n')
+    }))
+  ) {
+    await createFile(csvFileRealPath, '').catch((err) => {
+      throw prefixMessage(err, 'Error while creating CSV file.\n')
+    })
+  }
+
+  let csvContent = await readFile(csvFileRealPath).catch((err) => {
+    throw prefixMessage(err, 'Error while reading CSV file.\n')
+  })
+
+  let csvData = csvContent
+    .split('\n')
+    .filter((row) => row.length)
+    .map((data) => data.split(','))
+
+  const id = csvData.length === 0 ? 1 : csvData.length
+
+  csvData.push([id, ...data])
```

Comment:
This utility is assuming that first column will always be `id`, and auto-increment.

I think, it should be optional, else this utility cannot be used elsewhere with no such requirement.

Reply:
So, you are saying that we should have an optional argument `idColumn: boolean = false`. if `idColumn === true` then we should add id as first column otherwise not.

Reply:
It's column name is user-defined, it can be `id` `ID` `Id` `number` `no.` etc